src/core/Akka.Streams/Implementation/Sources.cs

```diff
@@ -749,4 +750,150 @@ public UnfoldResourceSourceAsync(Func<Task<TSource>> create, Func<TSource, Task<
         /// <returns>TBD</returns>
         public override string ToString() => "UnfoldResourceSourceAsync";
     }
+
+    /// <summary>
+    /// INTERNAL API
+    /// 
+    /// A graph stage that can be used to integrate Akka.Streams with .NET events.
+    /// </summary>
+    /// <typeparam name="TEventArgs"></typeparam>
+    /// <typeparam name="TDelegate">Delegate</typeparam>
+    public sealed class EventSourceStage<TDelegate, TEventArgs> : GraphStage<SourceShape<TEventArgs>>
+    {
+        #region logic
+
+        private class Logic : OutGraphStageLogic
+        {
+            private readonly EventSourceStage<TDelegate, TEventArgs> _stage;
+            private readonly LinkedList<TEventArgs> _buffer;
+            private readonly TDelegate _handler;
+            private readonly Action<TEventArgs> _onOverflow;
+
+            public Logic(EventSourceStage<TDelegate, TEventArgs> stage) : base(stage.Shape)
+            {
+                _stage = stage;
+                _buffer = new LinkedList<TEventArgs>();
+                var bufferCapacity = stage._maxBuffer;
+                var onEvent = GetAsyncCallback<TEventArgs>(e =>
```

Comment:
Right, my fault, didn't know that the function would be wrapped by the delegate, anyway lesson learned ;-)